src/overlays/actors/ovl_Bg_Ikana_Dharma/z_bg_ikana_dharma.c

```diff
@@ -28,47 +34,226 @@ const ActorInit Bg_Ikana_Dharma_InitVars = {
+        EffectSsKirakira_SpawnSmallYellow(globalCtx, &pos, &velocity, &accel);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Dharma/BgIkanaDharma_Init.s")
+void BgIkanaDharma_Init(Actor* thisx, GlobalContext* globalCtx) {
+    GlobalContext* globalCtx2 = globalCtx;
+    BgIkanaDharma* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Dharma/BgIkanaDharma_Destroy.s")
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    this->dyna.actor.scale.x = 0.3f;
+    this->dyna.actor.scale.y = 0.1f;
+    this->dyna.actor.scale.z = 0.3f;
+    DynaPolyActor_Init(&this->dyna, 0);
+    DynaPolyActor_LoadMesh(globalCtx, &this->dyna, &gStoneTowerTemplePunchablePillarCol);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->dyna.actor, &sCylinderInit);
+    if (!BGIKANADHARMA_IS_CHILD(&this->dyna.actor)) {
+        f32 segmentY = this->dyna.actor.world.pos.y;
+        s32 numSegments = BGIKANADHARMA_NUM_SEGMENTS(&this->dyna.actor);
+        s32 i;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Dharma/func_80BEC5C4.s")
+        for (i = 0; i < numSegments; i++) {
+            segmentY += 60.0f;
+            Actor_SpawnAsChildAndCutscene(&globalCtx2->actorCtx, globalCtx, ACTOR_BG_IKANA_DHARMA,
+                                          this->dyna.actor.world.pos.x, segmentY, this->dyna.actor.world.pos.z,
+                                          this->dyna.actor.shape.rot.x, this->dyna.actor.shape.rot.y,
+                                          this->dyna.actor.shape.rot.z, BGIKANADHARMA_FLAG_IS_CHILD,
+                                          this->dyna.actor.cutscene, this->dyna.actor.unk20, NULL);
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Dharma/func_80BEC5E0.s")
+        this->dyna.actor.bgCheckFlags |= 1;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Dharma/func_80BEC758.s")
+    BgIkanaDharma_BeginNormalState(this);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Dharma/func_80BEC790.s")
+void BgIkanaDharma_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgIkanaDharma* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Dharma/func_80BEC7EC.s")
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+    if (sFirstHitBgIkanaDharma == this) {
+        sFirstHitBgIkanaDharma = NULL;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Dharma/func_80BEC808.s")
+void BgIkanaDharma_BeginNormalState(BgIkanaDharma* this) {
+    this->actionFunc = BgIkanaDharma_UpdateNormalState;
+    this->dyna.actor.speedXZ = 0.0f;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Dharma/BgIkanaDharma_Update.s")
+void BgIkanaDharma_UpdateNormalState(BgIkanaDharma* this, GlobalContext* globalCtx) {
```

Comment:
I'd rather not call this Update since it's not an update function. Looks like it's just `Wait`, or maybe `WaitForHit`.